Tests for App

francistse23 opened the pull request:
#81 
- `npm run test-coverage` to run all the tests/tests suites and generate a report locally (add `/test-coverage` to `.gitignore` if you don't want to push this)

- Added tests for the app
- Achieving > 98% statements coverage
- Most components, utils, etc. have been accounted for

Kedyn commented on src/__tests__/colorConverter.test.ts:
```diff
@@ -0,0 +1,33 @@
+import { getColorValue, lightenDarkenColor } from "utils/colorConverter";
+
+import { COLOR_LIBRARY } from "constants/theme";
```
Never used.

Kedyn commented on src/__tests__/components/Player.test.tsx:
```diff
@@ -0,0 +1,107 @@
+import React, { KeyboardEvent } from "react";
```
KeyboardEvent never used.

Kedyn commented on src/__tests__/components/Sections.test.tsx:
```diff
@@ -0,0 +1,106 @@
+    beforeEach(async () => {
+      const mockStore = configureStore();
+      testStore = mockStore(store.getState());
+      dispatchSpy = jest.spyOn(testStore, "dispatch");
```
Assigned but never used.

Kedyn commented on src/utils/i18n.ts:
```diff
@@ -25,6 +25,11 @@ i18n
     },
+
+    react: {
+      useSuspense: (global as any).__DEV__ ? false : true,
```
lint gives a warning, but could not find a solution for the global type. Should add a ignore comment:
`// eslint-disable-line` or `// eslint-disable-next-line`.

Kedyn reviewed (approved):
Overall amazing work, just 4 minor linting warnings. Because these are only warnings and do not break anything, I do not think they really need to be addressed. So up to you if you want to address these warnings or just merge. Also, should we be deleting the test files within the components? Once again awesome stuff.

francistse23 commented on src/utils/AUMT/Layer.ts:
```diff
@@ -42,6 +43,14 @@ export default class Layer {
     return this.visible;
   }
 
+  public getEntities(): Array<Entity> {
+    return this.entities;
+  }
+
+  public setEntities(entities: Array<TextLine>): void {
+    this.entities = entities;
+  }
+
```
yeah was not too familiar with the whole public/private. just needed some way to access the variable to do the assertions

francistse23 commented on src/store/slices/PlayerEditLockSlice.ts:
```diff
@@ -10,16 +10,13 @@ const PlayerEditLockSlice = createSlice({
 
   togglePlayerEditLock,
-
-  resetPlayerEditLockState,
```
I don't recall making changes to this Slice specifically... let me check the main branch to see what it's supposed to be

francistse23 replied:
hmmm `git pull` didnt get that code for some reason. I added it back in

francistse23 commented:
thanks for pointing out the lint issues. addressed the issues and ran the command again. It should be free of issues now

also removed the unused test files since I grouped them all under `__tests__`

can be merged if you approve the changes
